### ISSUE TITLE
Feature Request: Two (or multiple) MAC addresses.

### ISSUE THREAD

**Issue by websafe** [snipe/snipe-it#1177]
Some assets, like laptops, routers, servers, have two (or more) network interfaces (LAN + WLAN, LAN + WAN, etc.) so I think it would be great to have the possibility to assign/add at least two MAC addresses to assets. Or maybe some kind of custom fields solution, that would allow to define custom fields for assets/accessories/licenses.

This issue looks like a duplicate of #877 but it's not, because IMHO IP addresses should not be stored in snipe-it and if - the IP addresses are assigned to MAC-addresses and depend on a specific network zone (for example a laptops NIC can have different IP addresses depending on the network segment to which the user is currently connected), and IP addresses are not assigned (only) to an asset, that's why the IP subject becomes very complex. 

So I would rather leave the IP addresses out for now and concentrate on an universal MAC address enhancement, where we can assign multiple MAC addresses to assets (accessories and licenses need MAC sometimes too), where we can select the type of NIC (wired|wireless) and other options: copper|fiber, WAN|LAN|Management and maybe a comment field, where one could store the IP address ;-)

Just one more thing. Licenses are in some cases assigned to a specific MAC address. The same goes for accessories like USB Bluetooth dongles, USB network interfaces, they also can have MAC addresses.

Peace & love

**Comment by snipe** [comment]
I think the custom fields that are being worked on by @uberbrady would render this obsolete.

**Comment by 4nd3r** [comment]
sorry for spamming in old issue, but what @websafe is describing is pretty similar to what i'm looking for. current customs fields solution doesn't allow easily to add multiple mac or IP-aadresses nor DNS names to assets.

with mac address I usually want to define respective IP-address and DNS name. in enterprise enviroments there's no dynamic DHCP IP-pools, but every device is precisely registered in asset management starting with MAC- and IP-aaddress down to switch port. also, same network interface can have multiple different IP-addresses depending on location (mobile employees who move between different offices, for example) or protocol version (ipv4, ipv6).

in summary: every device can have multiple network interfaces (lan, wlan, management). each interface has mac address. each mac address can have multiple IP-addresses and every IP-address can have multiple DNS names. also, there can be same DNS-name for multiple IP-addresses.

why is this all important? with all that information saved into Snipe-IT, one could automate updating of DHCP and DNS servers, switch port configurations and so on. of course Snipe-IT would need somekind of machine readable API for that to provide data for other systems.

so, this is my dream. cheers!

**Comment by snipe** [comment]
I'm still not clear on why you can't just use multiple custom fields here though.

**Comment by therealjoshuad** [comment]
That's how we do it. I have 4 custom fields (Wired MAC 1, Wired MAC 2, etc) that gets assigned to all server assets

Printers get a Wired Mac 1 and WiFi MAC, same goes for laptops, and desktops get wired mac 1

Edge cases go in the notes field

**Comment by 4nd3r** [comment]
@snipe current custom fields solution doesn't work for cases i described in summary
